script/tool/lib/src/common/xcode.dart

```diff
@@ -61,8 +77,36 @@ class Xcode {
     if (log) {
       print(completeTestCommand);
     }
-    return processRunner.runAndStream(_xcRunCommand, args,
+    final int resultExit = await processRunner.runAndStream(_xcRunCommand, args,
         workingDir: exampleDirectory);
+
+    if (resultExit != 0 && resultBundleTemp != null) {
+      final Directory xcresultBundle = resultBundleTemp.childDirectory(
+          'result.xcresult');
+      if (logsDirectory != null) {
+        if (xcresultBundle.existsSync()) {
+          // Zip the test results to the artifacts directory for upload.
+          final File zipPath = logsDirectory.childFile(
+              'xcodebuild-${DateTime.now().toLocal().toIso8601String()}.zip');
+          await processRunner.run(
+            'zip',
+            <String>[
+              '-r',
+              '-9',
+              '-q',
+              zipPath.path,
+              xcresultBundle.basename,
+            ],
+            workingDir: resultBundleTemp,
+          );
+        } else {
+          print('xcresult bundle ${xcresultBundle
+              .path} does not exist, skipping upload');
+        }
+      }
+    }
```

Comment:
Should we delete `resultBundleTemp` here so we aren't relying on CI to clean up `tmp/` regularly to avoid running out of space?